include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -259,8 +259,8 @@ class PreconditionerBoomerAMG : public PreconditionerBase<Number>
   BoomerData boomer_data;
 
   // PETSc vector objects to avoid re-allocation in every vmult() operation
-  mutable VectorTypePETSc petsc_vector_src;
-  mutable VectorTypePETSc petsc_vector_dst;
+  mutable Vec petsc_vector_src;
+  mutable Vec petsc_vector_dst;
```

Comment:
Is there a reason you change this? I think we introduced the more verbose name https://github.com/exadg/exadg/blob/c62fd0a241beab24a91cea6fd64680cbfb2c17cd/include/exadg/solvers_and_preconditioners/utilities/petsc_operation.h#L31-L34 to ensure that the ambiguous name `Vec` is not ending up all over the place. We could move that to a more central location in ExaDG if you do not want to include `petsc_operator.h` in `operator_base.h`, or some similar setting.

Reply:
OK, let's keep this for now. For matrices one can probably create a simple interface with the basic methods to add entries, mat-vec and similar. The problem is that interfaces to deal.II are going to be a bit harder to adjust because several of those are templated, rather than accessed via virtual functions. One could see if there is a way to make that uniform there, but it would probably be a bigger project.

Reply:
I tried the variant above (`ExaDG::SparseMatrix`) but did not find a reasonable solution to hide the ifdef's in `OperatorBase` and similar files/classes. The problem is in my opinion the circumstance that the classes `MultigridOperatorBase` and `MultigridOperator` use virtual functions where we cannot use a template for the sparse matrix (for this reason, also these two classes implement certain functions for Trilinos and PETSc in the current code). `ExaDG::SparseMatrix` worked well just for the `vmult()` function, which is however only one of several functions where we distinguish between Trilinos and PETSc, which is why I think it is not worth introducing `ExaDG::SparseMatrix` just for this function and still have the ifdefs for the other occurrences.

It could work better if we derive `ExaDG::SparseMatrix` from a sparse matrix type/class in deal.II and then integrate multiple sparse matrix types (like Trilinos, PETSc) via composition and ifdef's in the member functions of such a class. The funtions `internal_init/calculate_system_matrix()` in `OperatorBase` currently use a template `SparseMatrix`, meaning that we rely on a common interface that both Trilinos and PETSc fulfill. Hence, deriving `ExaDG::SparseMatrix` from a deal.II sparse matrix fulfilling this interface could imply that we can just remove these template parameters from certain functions in `OperatorBase`, `MultigridOperatorBase`, and `MultigridOperator`. However, I did not try this, so potential problems (which I am not aware of right now) might show up there as well.